lib/Sema/ConstraintGraph.h

```diff
@@ -222,10 +180,11 @@ class ConstraintGraph {
   /// Describes which constraints \c gatherConstraints should gather.
   enum class GatheringKind {
     /// Gather constraints associated with all of the variables within the
-    /// same equivalence class as the given type variable.
+    /// same equivalence class as the given type variable, as well as its
+    /// immediate fixed bindings.
     EquivalenceClass,
     /// Gather all constraints that mention this type variable or type variables
-    /// that it is equivalent to.
+    /// that it is a fixed binding of.
```

Comment:
I think it might be worth to clarify this comment to describe exactly what is going to be covered in this case in comparison to `EquivalenceClass` and when this should be used...

Reply:
@xedin I've gone ahead and added a bit more to the comment, how does it look?

I've also changed the unsolved constraint assert into a `llvm::report_fatal_error` to make sure we don't end up in situations where we have an unsolved constraint along one branch, but not another branch, and just quietly take the other branch in a no-asserts build.

Thanks for the review!

Reply:
Looks great, thank you!